docs/source/caching/programmatic-ids.mdx

```diff
@@ -59,25 +52,27 @@ val objectIdGenerator = object : CacheKeyGenerator {
 }
 ```
 
-Note how this requires `id` to be present in the response. If a query does not return this field, the `CacheKey` will be inconsistent and duplication will happen.
+You can also use the current object's typename to use _different_ cache ID generation logic for different object types.
```

Comment:
Is this accurate? I assume this function is getting called for every object type, and you might need to `switch` on the typename or similar to use different logic per type

Reply:
Pardon my kotlin ignorance. `when` 😄

Reply:
Yes, that's correct 👍 It's called for every object.